regulation-worker/internal/delete/api/api.go

```diff
@@ -223,9 +224,9 @@ func (api *APIManager) refreshOAuthToken(destName, workspaceId string, oAuthDeta
 		DestDefName:     destName,
 		EventNamePrefix: "refresh_token",
 	}
-	statusCode, _ := api.OAuth.RefreshToken(refTokenParams)
+	statusCode, refreshResponse := api.OAuth.RefreshToken(refTokenParams)
```

Comment:
`refreshResponse` is a pointer, can't it be `nil`?

Reply:
We are some or other response through `RefreshToken`. Except in one-case where while we're trying to marshal to send request to control-plane, we are panicking there. [Here is the source-code reference](https://github.com/rudderlabs/rudder-server/blob/fix.delete-users-oauth/services/oauth/oauth.go#L298-L301)

Reply:
would it be too much to add a nil check here?

Reply:
We can add it, not a problem 👍🏼

Reply:
Added it